FastPriorityQueue.js

```diff
@@ -287,13 +287,22 @@ FastPriorityQueue.prototype.forEach = function(callback) {
 FastPriorityQueue.prototype.kSmallest = function(k) {
   if (this.size == 0) return [];
```

Comment:
```suggestion
  if ((this.size == 0) || (k<=0)) return [];
```

Reply:
I implemented this change after testing. Similar checks I had tested at this point degraded performance, but this did not.